[ONNX] Update the default opset to version 14

justinchuby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83284
* #83283

Update the default opset by running the `update_default_opset_version.py` script. The update is done in a regularly to ensure we are in sync with the onnx updates. All changes are produced by the script.

justinchuby committed:
[ONNX] Update the default opset

Update the default opset by running the `update_default_opset_version.py` script

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Update the default opset"


Update the default opset by running the `update_default_opset_version.py` script

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Update the default opset"


Update the default opset by running the `update_default_opset_version.py` script

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Update the default opset"


Update the default opset by running the `update_default_opset_version.py` script

[ghstack-poisoned]

thiagocrepaldi reviewed (commented):
Hi @justinchuby, please provide a clear and concise description of why we are changing the default opset. Documenting the reason helps us in the future, when we are debugging problems that are specific to opset version that affects a particular runtime

What is the motivation for the proposal? Changing the default opset can break Caffe2 and any other runtime that consumes. IIRC, it might be because we are trying to stay 3 version behind the latest, is that it?

thiagocrepaldi commented on test/onnx/expect/TestOperators.test_view.expect:
```diff
@@ -22,6 +22,11 @@ graph {
     output: "2"
     name: "Reshape_1"
     op_type: "Reshape"
+    attribute {
+      name: "allowzero"
+      i: 0
+      type: INT
+    }
```
are these expected with opset 14?

justinchuby replied:
I think so? They are automatically generated.

justinchuby replied:
I found the documented change in https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Reshape-14

justinchuby replied:
Marking as resolved

justinchuby commented:
> Hi @justinchuby, please provide a clear and concise description of why we are changing the default opset. Documenting the reason helps us in the future, when we are debugging problems that are specific to opset version that affects a particular runtime
> 
> What is the motivation for the proposal? Changing the default opset can break Caffe2 and any other runtime that consumes. IIRC, it might be because we are trying to stay 3 version behind the latest, is that it?

Done. PTAL.

The change is done by running the script. As I understand it - yes it is because we want to use the oldest version in the last 18 months (mentioned in the script).

If versions are usually explicitly specified during conversion, I am hoping nothing will break with this change?

thiagocrepaldi commented:
> > Hi @justinchuby, please provide a clear and concise description of why we are changing the default opset. Documenting the reason helps us in the future, when we are debugging problems that are specific to opset version that affects a particular runtime
> > What is the motivation for the proposal? Changing the default opset can break Caffe2 and any other runtime that consumes. IIRC, it might be because we are trying to stay 3 version behind the latest, is that it?
> 
> Done. PTAL.
> 
> The change is done by running the script. As I understand it - yes it is because we want to use the oldest version in the last 18 months (mentioned in the script).
> 
> If versions are usually explicitly specified during conversion, I am hoping nothing will break with this change? Do we need an import to verify?

Assuming people use explicit version on the torch.onnx.export call, nothing should break, but the API also allows not setting it, so a break is possible. Although the ONNX converter team know about the policy for updating the opset, the general public probably dont, so expliciting the reason on the PR is always a good idea. Especially for those that do not explicitly specify opset version and spend days debugging a model change they have no idea where it came from :P

thiagocrepaldi reviewed (commented):
The change LGTM.

One aspect that we never discussed when changing opset versions before, but I thought about it now is: Should we add `module: BC-breaking` or `topic: BC breaking` Github labels on the PR that changes opset version? 

For scripts that use implicit opset version, the opset change can cause BC issues and the PyTorch release notes should explicit that

@BowenBao @shubhambhokare1 any thoughts?

BowenBao commented:
> The change LGTM.
> 
> One aspect that we never discussed when changing opset versions before, but I thought about it now is: Should we add `module: BC-breaking` or `topic: BC breaking` Github labels on the PR that changes opset version?
> 
> For scripts that use implicit opset version, the opset change can cause BC issues and the PyTorch release notes should explicit that
> 
> @BowenBao @shubhambhokare1 any thoughts?

I agree with the BC breaking concern and I think it's the cost we have to pay for moving forward. Good idea to add the BC-breaking tags. Linking the original PR from Gary for reference #73898.

@justinchuby I'm not sure about this line `Update the default opset ... to prepare for opset 17 support`. Why are these related? If there is no dependency I'd actually recommend separating this PR, as it might get reverted for failing Meta internal tests.

justinchuby commented:
> I agree with the BC breaking concern and I think it's the cost we have to pay for moving forward. Good idea to add the BC-breaking tags. Linking the original PR from Gary for reference #73898.
> 
> @justinchuby I'm not sure about this line `Update the default opset ... to prepare for opset 17 support`. Why are these related? If there is no dependency I'd actually recommend separating this PR, as it might get reverted for failing Meta internal tests.

Thanks for pointing this out. For some reason I just assumed it is the process when there is a new opset coming out. I have detached #83287

justinchuby committed:
Update on "[ONNX] Update the default opset"


Update the default opset by running the `update_default_opset_version.py` script. The update is done in a regularly to ensure we are in sync with the onnx updates. All changes are produced by the script.

[ghstack-poisoned]

titaiwangms reviewed (approved):
🍺

justinchuby committed:
Update on "[ONNX] Update the default opset"


Update the default opset by running the `update_default_opset_version.py` script. The update is done in a regularly to ensure we are in sync with the onnx updates. All changes are produced by the script.

[ghstack-poisoned]

malfet reviewed (approved):
LGTM, not sure why it has `onnx-needs-import` label

justinchuby commented:
@pytorchbot merge -g

justinchuby commented:
@pytorchbot merge -f "unrelated xla build error"
